pkg/core/diagnostics.go

```diff
@@ -10,32 +10,14 @@ func (d *ConfigureDiagnostic) IsConfigureDiagnostic() bool {
 	return true
 }
 
-func convertToConfigureDiagnostics(dd diag.Diagnostics) diag.Diagnostics {
-	ret := make(diag.Diagnostics, len(dd))
-	for i := range dd {
-		ret[i] = &ConfigureDiagnostic{
-			Diagnostic: dd[i],
-		}
-	}
-	return ret
-}
-
 type FetchDiagnostic struct {
 	diag.Diagnostic
 	Provider string
 	Version  string
 }
 
-func convertToFetchDiags(diags diag.Diagnostics, provider, version string) diag.Diagnostics {
-	fd := make(diag.Diagnostics, len(diags))
-	for i, d := range diags {
-		fd[i] = FetchDiagnostic{
-			Diagnostic: d,
-			Provider:   provider,
-			Version:    version,
-		}
-	}
-	return fd
+func (d *FetchDiagnostic) IsFetchDiagnostic() (bool, string, string) {
+	return true, d.Provider, d.Version
```

Comment:
This is not very scalable as we should stick to base types (to avoid importing this pkg from errors)

Reply:
Fixed in [1c11f31](https://github.com/cloudquery/cloudquery/pull/744/commits/1c11f3130a446a99761ee4b072da8239fd288b18)